packages/cadl-ranch-specs/http/azure/core/page/main.tsp

```diff
@@ -200,3 +201,411 @@ interface TwoModelsAsPageItem {
+   */
+  @header
+  ETag?: string;
+
+  /**
+   * The time at which the resource was last modified.
+   */
+  @header
+  @encode(DateTimeKnownEncoding.rfc7231)
+  `Last-Modified`?: utcDateTime;
+}
+
+/**
+ * An error response received from the Azure Batch service.
+ */
+@error
+model BatchError {
+  /**
+   * An identifier for the error. Codes are invariant and are intended to be consumed programmatically.
+   */
+  code: string;
+}
+
+@scenario
+@scenarioDoc("""
+  Lists all of the Pools in the specified Account.
+  """)
+interface Pools {
+  /**
+   * Lists all of the Pools in the specified Account.
+   */
+  @route("/pools")
+  @put
+  listPools is global.Azure.Core.RpcOperation<
+    BatchRequestHeaders,
+    BatchPoolListResult,
+    BatchResponseHeaders,
+    BatchError
+  >;
+}
+
+/**
+ * Text Blocklist.
+ */
+@resource("blocklists")
+model TextBlocklist {
+  /**
+   * The name of the text blocklist. This can only include the following characters: 0-9 A-Z a-z - . _ ~
+   */
+  @key("blocklistName")
+  @visibility("read", "create", "query")
+  @maxLength(64)
+  blocklistName: string;
+
+  /**
+   * A description of the text blocklist.
+   */
+  @maxLength(1024)
+  description?: string;
+}
+
+/**
+ * Item in TextBlocklist.
+ */
+@resource("blockItems")
+@parentResource(TextBlocklist)
+model TextBlockItem {
+  /**
+   * The unique identifier for the block item.
+   */
+  @key("blockItemId")
+  @visibility("read", "create", "query")
+  @maxLength(64)
+  blockItemId: string;
+
+  /**
+   * A description of the block item.
+   */
+  @maxLength(1024)
+  description?: string;
+
+  /**
+   * The content of the block item.
+   */
+  @maxLength(128)
+  text: string;
+}
+
+@scenario
+@scenarioDoc("""
+  This scenario is to demonstrate an interface with two operations, each returning a list of different resource types.
+  """)
+interface TwoResourcesAsListItems {
+  /**
+   * Get all text blocklists details.
+   */
+  @route("/Text-blocklists")
+  listTextBlocklists is global.Azure.Core.ResourceList<TextBlocklist>;
+
+  /**
+   * Get all blockItems in a text blocklist
+   */
+  @route("/Text-blockItems")
+  listTextBlocklistItems is global.Azure.Core.ResourceList<
+    TextBlockItem,
+    global.Azure.Core.Traits.ListQueryParametersTrait<global.Azure.Core.StandardListQueryParameters>
+  >;
+}
+
+/**
+ * User roles
+ */
+union LedgerUserRole {
+  string,
+  Administrator: "Administrator",
+  Contributor: "Contributor",
+  Reader: "Reader",
+}
+
+@scenario
+@scenarioDoc("""
+  This operation is used to list custom resources.
+  """)
+@route("/custom-resource-list")
+op ListCustomResource<
+  TResource extends {},
+  TCustom extends global.Azure.Core.Foundations.CustomizationFields = {}
+> is global.Azure.Core.Foundations.ResourceList<
+  TResource,
+  global.Azure.Core.Foundations.CustomParameters<TCustom>,
+  global.Azure.Core.Page<TResource>
+>;
```

Comment:
if this template is only used in one place, we do not need to have it. It reduces the readability.
Please inline it or we could try to make abstraction to all the new operations into one template - I see all the new operations are trying to list a resource, maybe we could do it.